### PR TITLE
Purchase tickets improvements

### PR DESCRIPTION
**Pull request by guilhermemntt**
Closes https://github.com/decred/decrediton/issues/2303.

### CODE REVIEW

**Comment by guilhermemntt** [comment]
I don't believe I haven't noted the wrong icon there, as well the ``yarn-error.log``, my bad. Concerning the maintained rules, other components depend on them. The same issue occurs with the amount on the purchase ticket form. If I change it to making the unit label be closely to the unit number, other inputs along the application will break, that's the reason why I had to put an ``!important`` on it.

File: app/style/themes/darkTheme.js [hunk 1]
```diff
@@ -218,8 +218,14 @@ const darkTheme = {
   "menu-trezor-active": "url('style/icons/trezorActiveDark.svg')",
   "menu-trezor-default": "url('style/icons/trezorDefaultDark.svg')",
   "menu-trezor-hover": "url('style/icons/trezorHoverDark.svg')",
-  "menu-cancel-rescan-icon": "url('style/icons/menuCancelRescanDark.svg')"
-
+  "menu-cancel-rescan-icon": "url('style/icons/menuCancelRescanDark.svg')",
```
Comment (amassarwi) [inline app/style/themes/darkTheme.js:221]: Please move these to `themes/icons.js` => it's the shared icons file, here we specify dark specific icons, same hold for icons in `lightTheme.js`

File: app/actions/ClientActions.js [hunk 2]
```diff
@@ -332,7 +332,9 @@ export const getStakeInfoAttempt = () => (dispatch, getState) => {
   dispatch({ type: GETSTAKEINFO_ATTEMPT });
   wallet.getStakeInfo(sel.walletService(getState()))
     .then(resp => {
+
```
Comment (amassarwi) [inline app/actions/ClientActions.js:335]: Extra 2 empty lines

File: app/style/themes/lightTheme.js [hunk 3]
```diff
@@ -217,7 +217,14 @@ const lightTheme = {
   "menu-trezor-active": "url('style/icons/trezorActive.svg')",
   "menu-trezor-default": "url('style/icons/trezorDefault.svg')",
   "menu-trezor-hover": "url('style/icons/trezorHover.svg')",
-  "menu-cancel-rescan-icon": "url('style/icons/menu-cancel-rescan.svg')"
+  "menu-cancel-rescan-icon": "url('style/icons/menu-cancel-rescan.svg')",
```
Comment (amassarwi) [inline app/style/themes/lightTheme.js:220]: See comment on `darktTheme.js`

File: app/style/themes/lightTheme.js [hunk 4]
```diff
@@ -219,8 +219,7 @@ const lightTheme = {
   "info-icon": "url('style/icons/information.svg')",
   "menu-trezor-active": "url('style/icons/trezorActive.svg')",
   "menu-trezor-default": "url('style/icons/trezorDefault.svg')",
-  "menu-trezor-hover": "url('style/icons/trezorHover.svg')",
-  "menu-cancel-rescan-icon": "url('style/icons/menu-cancel-rescan.svg')"
```
Comment (amassarwi) [inline app/style/themes/lightTheme.js:223]: shouldn't be deleted

File: app/style/themes/darkTheme.js [hunk 5]
```diff
@@ -220,8 +220,7 @@ const darkTheme = {
   "info-icon": "url('style/icons/informationDark.svg')",
   "menu-trezor-active": "url('style/icons/trezorActiveDark.svg')",
   "menu-trezor-default": "url('style/icons/trezorDefaultDark.svg')",
-  "menu-trezor-hover": "url('style/icons/trezorHoverDark.svg')",
-  "menu-cancel-rescan-icon": "url('style/icons/menuCancelRescanDark.svg')"
```
Comment (amassarwi) [inline app/style/themes/darkTheme.js:224]: shouldn't be deleted

File: app/style/themes/lightTheme.js (hunk 4 above)
Comment (guilhermemntt) [reply app/style/themes/lightTheme.js]: Fixed.

File: app/style/themes/darkTheme.js (hunk 5 above)
Comment (guilhermemntt) [reply app/style/themes/darkTheme.js]: Fixed.

File: app/reducers/grpc.js [hunk 6]
```diff
@@ -383,7 +383,23 @@ export default function grpc(state = {}, action) {
         getTicketsRequestAttempt: false,
         getTicketsStartRequestHeight: action.getTicketsStartRequestHeight,
         getTicketsCancel: false,
-        getTicketsProgressStartRequestHeight: null
+        getTicketsProgressStartRequestHeight: null,
+        lastTicket: (() => {
```
Comment (vctt94) [inline app/reducers/grpc.js:387]: Move this logic to selector, instead of here

File: app/selectors.js [hunk 7]
```diff
@@ -999,6 +999,8 @@ export const totalSubsidy = compose(
   getStakeInfoResponse
 );
 
+export const lastTicket = get(["grpc", "lastTicket"]);
```
Comment (vctt94) [inline app/selectors.js:1002]: can be renamed to `lastVotedTicket`

**Comment by guilhermemntt** [comment]
Rebased.

File: app/selectors.js [hunk 8]
```diff
@@ -703,6 +703,20 @@ export const filteredStakeTxs = createSelector(
   }
 );
 
+export const lastVotedTicket = createSelector(
+  [stakeTransactions],
+  (transactions) => {
+    const lastVotedTicket = Object.keys(transactions)
+      .map((hash) => transactions[hash])
+      .filter((transaction) => transaction.status=="voted")
```
Comment (vctt94) [inline app/selectors.js:711]: Use our constant from `constants/Decrediton.js`

Comment (guilhermemntt) [reply app/selectors.js]: Fixed. Thanks for the suggestion!

**Pull request merged**